themes/default/content/registry/packages/aws/how-to-guides/bucketv2-migration.md

```diff
@@ -0,0 +1,3701 @@
+---
+title: "BucketV2 Migration Guide"
+h1: "Migrating from aws.s3.Bucket to aws.s3.BucketV2"
+meta_desc: Practitioner level instructions for migrating from aws.s3.Bucket to aws.s3.BucketV2 resources
+layout: package
+---
+
+In the upcoming AWS Classic major release (v7), `aws.s3.Bucket` will be discontinued in favor of `BucketV2`. Users of
+`aws.s3.Bucket` resource are encouraged to migrate early. The migration requires a significant refactor to the source
+code and additional steps for avoiding data loss. This guide aims to cover all relevant scenarios with precise migration
+instructions.
+
+To migrate existing `aws.s3.Bucket` resources to `aws.s3.BucketV2`:
+
+1. Edit your Pulumi program sources to replace removed inputs with equivalent side-by-side resources. Specifically
+   `BucketV2` inputs such as `policy` and `accelerationStatus` are to be replaced by side-by-side resources
+   `aws.s3.BucketPolicy` and `aws.s3.BucketAccelerateConfigurationV2`.
+
+2. Perform `pulumi up` to replace the buckets in AWS. **WARNING**: replacing the buckets will delete and re-create them,
```

Comment:
I'm wondering if we should flip this and mention replacing buckets as the alternative. I think 99% of buckets won't be able to be replaced so we should just go with that as the default.

Reply:
You're right.. that seems to be really true. I'll incorporate.